Paintroid/src/main/java/org/catrobat/paintroid/model/LayerModel.kt

```diff
@@ -22,25 +22,32 @@ import android.graphics.Bitmap
 import android.graphics.Canvas
 import org.catrobat.paintroid.contract.LayerContracts
 import java.util.ArrayList
+import kotlin.IndexOutOfBoundsException
 
 class LayerModel : LayerContracts.Model {
     override var currentLayer: LayerContracts.Layer? = null
     override var width = 0
     override var height = 0
-    override val layers: ArrayList<LayerContracts.Layer> = ArrayList()
+    override val layers: ArrayList<LayerContracts.Layer> = arrayListOf()
     override val layerCount: Int
         get() = layers.size
 
     override fun reset() {
         layers.clear()
     }
 
-    override fun getLayerAt(index: Int): LayerContracts.Layer = layers[index]
+    override fun getLayerAt(index: Int): LayerContracts.Layer? = layers.getOrNull(index)
 
     override fun getLayerIndexOf(layer: LayerContracts.Layer): Int = layers.indexOf(layer)
 
-    override fun addLayerAt(index: Int, layer: LayerContracts.Layer) {
-        layers.add(index, layer)
+    @SuppressWarnings("SwallowedException", "TooGenericExceptionCaught")
+    override fun addLayerAt(index: Int, layer: LayerContracts.Layer): Boolean {
+        return try {
+            layers.add(index, layer)
+            true
+        } catch (e: IndexOutOfBoundsException) {
+            false
+        }
     }
```

Comment:
```suggestion
    @SuppressWarnings("SwallowedException", "TooGenericExceptionCaught")
    override fun addLayerAt(index: Int, layer: LayerContracts.Layer): Boolean = try {
        layers.add(index, layer)
        true
    } catch (e: IndexOutOfBoundsException) {
        false
    }
```
You can use expression body syntax here

Reply:
good idea, missed it